src/app/meshes/views/MeshView.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <KTabs
```

Comment:
**Issue**: As `KTabs` isn’t built with links in mind and so its styles don’t accommodate that use well: currently, the clickable area of the link is just the box of the text but not the padding box of the tab making only the text behave like a link.

We will need to override `KTabs` and make it so that the link takes up the whole space of the tab for it to work correctly.

Is this also what the `click` event handler on `KTabs` is for?

Reply:
🤔 yeah kinda between a rock and a hard place here, not totally sure whats best long term, umm I'm gonna change things up to loop through the tabs dynamically, so I might take another look at this also, not totally sure.

Reply:
I took another look at this whilst doing the tab loop-de-loop thing. And I figured how I've left it right now is the least intrusive.

- The CSS I'm using isn't doing any overwrites and is only effecting slots which I control
- The CSS only affects colors, and the colors I'm using are the KTab variable colors, so thats pretty safe as an overwrite.
- There is one more overwrite for removing underlines, and I know there is an issue (https://github.com/kumahq/kuma-gui/issues/738) which will make this unnecessary pretty soon.

The JS approach I'm using is a little weird considering, but there are no overwrites and it's pretty acceptable - just that its a bit weird that we have to add this to do anchored tabs but no biggie really. I'm pretty sure that any changes to KTabs wouldn't break this unless explicitly stopped bubbling events.

Reply:
In my opinion this is functionally broken. Buttons are not links and making this work correctly requires minimal overriding of Kongponents styles which is far less problematic than fake link activation triggered via `click` event. Below is what I just tried locally to make the clickable area as big as the tab:

In `src/app/meshes/views/MeshView.vue`:

- Add class `nav-tabs` to `KTabs`
- Add class `nav-tab-link` to `router-link`
- Add the following rule set to the scoped style block:

  ```scss
  .nav-tab-link {
    display: block;
    padding: var(--spacing-md);
  }
  ```

- Add the following unscoped style block:

  ```vue
  <style lang="scss">
  .nav-tabs .tab-item {
    padding: 0 !important;
  }
  </style>
  ```

Reply:
Not keen on the unscoped CSS block with an `!important` hiding in a component somewhere deep the app folder structure but 🤷 , as long as we don't use `.nav-tabs` elsewhere.

Oh actually:

![Screenshot 2023-05-19 at 10 13 42](https://github.com/kumahq/kuma-gui/assets/554604/26e5bf55-5341-4193-bac5-0a1646d474a1)

I suppose this could be fixed with a `.nav-tab > ul .tab-item`? But it makes me think I'd rather stick with standard DOM event approach for the moment just as we know for sure it doesn't affect anything else like the global `!important` might.

Lemme know. If you are still super strong on the global `!important` CSS rule then I could fix this up with a `>`. I'm still not that precious about either approach really.

Reply:
> I suppose this could be fixed with a .nav-tab > ul .tab-item?

This would indeed avoid that issue.

> But it makes me think I'd rather stick with standard DOM event approach for the moment just as we know for sure it doesn't affect anything else like the global !important might.

For what it’s worth, you could remove the `!important` by increasing the selector’s specificity sufficiently (e.g. `.nav-tabs > ul > .tab-item:not(.increase-specificity)` would do the trick).

I definitely want this addressed. Triggering a click via JavaScript isn’t enough to make the tab behave like a link.

Reply:
> Triggering a click via JavaScript isn’t enough to make the tab behave like a link.

What's the issue here?

Reply:
Good points, I switched to CSS but used a classname that is highly unlikely to be reused elsewhere.